lightstep/resource_metric_dashboard.go

```diff
@@ -561,50 +613,14 @@ func (p *resourceUnifiedDashboardImp) setResourceDataFromUnifiedDashboard(projec
 		return fmt.Errorf("unable to set type resource field: %v", err)
 	}
 
-	assembleCharts := func(chartsIn []client.UnifiedChart) ([]interface{}, error) {
-		var charts []interface{}
-		for _, c := range chartsIn {
-			chart := map[string]interface{}{}
-
-			yMap := map[string]interface{}{}
-
-			if c.YAxis != nil {
-				yMap["max"] = c.YAxis.Max
-				yMap["min"] = c.YAxis.Min
-				chart["y_axis"] = []map[string]interface{}{yMap}
-			}
-
-			if p.chartSchemaType == MetricChartSchema {
-				chart["query"] = getQueriesFromMetricConditionData(c.MetricQueries)
-			} else {
-				queries, err := getQueriesFromUnifiedDashboardResourceData(
-					c.MetricQueries,
-					dash.ID,
-					c.ID,
-				)
-				if err != nil {
-					return nil, err
-				}
-				chart["query"] = queries
-			}
-			chart["name"] = c.Title
-			chart["rank"] = c.Rank
-			chart["x_pos"] = c.Position.XPos
-			chart["y_pos"] = c.Position.YPos
-			chart["width"] = c.Position.Width
-			chart["height"] = c.Position.Height
-			chart["type"] = c.ChartType
-			chart["id"] = c.ID
-
-			charts = append(charts, chart)
-		}
-		return charts, nil
-	}
 	if isLegacyImplicitGroup(dash.Attributes.Groups, hasLegacyChartsIn) {
-		charts, err := assembleCharts(dash.Attributes.Groups[0].Charts)
+		charts, textPanels, err := assembleDashboardPanels(dash.ID, p.chartSchemaType, dash.Attributes.Groups[0].Charts)
 		if err != nil {
 			return err
 		}
+		if len(textPanels) > 0 {
+			return fmt.Errorf("text panels are only supported within groups")
```

Comment:
AFAIK, we don't intend to support text panels outside of groups as everything is intended to be in groups moving forward.